src/pages/sitemap-docs.xml.tsx

```diff
@@ -0,0 +1,31 @@
+import { GetServerSidePropsContext } from 'next'
+
+export default function SiteMap() {
+	return null
+}
+
+const siteUrl = process.env.SITE_URL || 'https://developer.hashicorp.com'
+
+export async function getServerSideProps({ res }: GetServerSidePropsContext) {
+	const response = await fetch(
+		`https://content.hashicorp.com/api/all-docs-paths`
+	)
+	const { result } = await response.json()
+	const createSitemap = () => `<?xml version="1.0" encoding="UTF-8"?>
+    <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+        ${result
+					.map(
+						(sitePath: { path: string; created_at: string }) =>
+							`<url><loc>${siteUrl}/${sitePath.path}</loc><lastmod>${sitePath.created_at}</lastmod><changefreq>daily</changefreq><priority>1.0</priority></url>
+              `
+					)
+					.join('')}
+    </urlset>
+    `
```

Comment:
Did you see [this method](https://github.com/iamvishnusankar/next-sitemap#server-side-sitemap-getserversidesitemap) exported from `next-sitemap`? It seems like it could be helpful here.